exporter.go

```diff
@@ -252,48 +216,50 @@ func main() {
 
 	http.Handle(*metricsPath, promhttp.Handler())
 
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		_, err := fmt.Fprintf(w, `<!DOCTYPE html>
-			<title>NGINX Exporter</title>
-			<h1>NGINX Exporter</h1>
-			<p><a href=%q>Metrics</a></p>`,
-			*metricsPath)
+	if *metricsPath != "/" && *metricsPath != "" {
+		landingConfig := web.LandingConfig{
+			Name:        "NGINX Prometheus Exporter",
+			Description: "Prometheus Exporter for NGINX and NGINX Plus",
+			HeaderColor: "#039900",
+			Version:     version.Info(),
+			Links: []web.LandingLinks{
+				{
+					Address: *metricsPath,
+					Text:    "Metrics",
+				},
+			},
+		}
+		landingPage, err := web.NewLandingPage(landingConfig)
 		if err != nil {
-			level.Error(logger).Log("msg", "Error while sending a response for the '/' path", "error", err.Error())
+			level.Error(logger).Log("err", err)
 			os.Exit(1)
 		}
-	})
+		http.Handle("/", landingPage)
+	}
 
-	listener, err := getListener(*listenAddr)
-	if err != nil {
-		level.Error(logger).Log("msg", "Could not create listener", "error", err.Error())
-		os.Exit(1)
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
```

Comment:
IMHO it should be `syscall.SIGTERM` instead of `os.Kill`

Without this you don't hit line 259 ("Shutting down")

https://www.gnu.org/software/libc/manual/html_node/Termination-Signals.html#index-SIGKILL

Reply:
I didn't notice you updated the comment before I posted basically the same 😅